paper-drawer-panel.html

```diff
@@ -834,6 +835,18 @@
 
         _isDrawerClosed: function(narrow, selected) {
           return !narrow || selected !== 'drawer';
+        },
+
+        _rightDrawerChanged: function() {
+          var transition = this._transition;
+          this._transition = false;
+          this._rightDrawer = this.rightDrawer;
+          this._forceReflow(this.$.drawer);
```

Comment:
This is bad for performance. You can also set `this._transition=transition` in a `requestAnimationFrame` block.  Also, you can check the old value of `rightDrawer` from `_rightDrawerChanged: function(newRightDrawer, oldRightDrawer)` and if it's `undefined`, you can can skip doing all this setup.